node/node.go

```diff
@@ -24,7 +24,7 @@ var log = logging.Logger("node")
 // * Light
 type Node struct {
 	Type   Type
-	Config *Config
+	Config ConfigLoader
```

Comment:
Why? What's the benefit here?

Reply:
Looks like this is necessary to easily match that interface here? 
https://github.com/celestiaorg/celestia-node/blob/3047a265eaf7634ea0838af4e4f9192371129212/node/repo.go#L40-L41

Not sure if it justifies creating a new public type though or in which scenarios a `ConfigLoader` adds value.

Reply:
@renaynay, his is the right question. The motivation here is behind having the ability to load the latest config version from a disk. As a node operator, I guarantee that this will become useful. So for now we have the initial config and the ability to load it from disk ad-hoc. The more complicated approach is to have some FS listener(fsnotify) that would update the initial config on any config file update.

Reply:
> Let's just load the config during initialisation and store the pointer to the config on the Node please

So we have the initial Config and ConfigLoader, I just omitted the initial config from the Node, but can add it for visibility.

Reply:
Resolved with @renaynay in sync